components/homme/src/share/sl_advection.F90

```diff
@@ -937,9 +937,9 @@ subroutine calc_vertically_lagrangian_levels( &
 
     dp_neg_min = reconstruct_and_limit_dp(elem%state%dp3d(:,:,:,tl%np1), &
          dt, dp_tol, eta_dot_dpdn(:,:,:,1), dprecon)
-#ifndef NDEBUG
+#if 0
```

Comment:
Why not leaving the existing ifndef?

Reply:
There was an issue the other day where E3SM's CMake system accidentally removed -DNDEBUG in optimized builds, and that output confused some folks running simulations. It's pretty deep in the weeds. I tend to enable it manually when I want to examine certain things that transport can feel. Since it's not something that benefits others -- and leads to confusion -- I'm removing it from standard builds.